File: Codes/2535-difference-between-element-sum-and-digit-sum-of-an-array.py
'''
Author: LetMeFly
Date: 2024-09-26 18:07:21
LastEditors: LetMeFly
LastEditTime: 2024-09-26 18:08:59
'''
from typing import List

class Solution:
    def getSum(self, x: int) -> int:
        ans = 0
        while x:
            ans += x % 10
            x //= 10
        return ans

    def differenceOfSum(self, nums: List[int]) -> int:
        x = y = 0
        for t in nums:
            x += t
            y += self.getSum(t)
        return abs(x - y)

File: Codes/2535-difference-between-element-sum-and-digit-sum-of-an-array.java
/*
 * @Author: LetMeFly
 * @Date: 2024-09-26 18:07:24
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-09-26 18:10:12
 */
class Solution {
    private int getSum(int t) {
        int ans = 0;
        while (t > 0) {
            ans += t % 10;
            t /= 10;
        }
        return ans;
    }

    public int differenceOfSum(int[] nums) {
        int x = 0, y = 0;
        for (int t : nums) {
            x += t;
            y += getSum(t);
        }
        return Math.abs(x - y);
    }
}